### ISSUE TITLE
Allow to set `time_speed` programmatically

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#14922]
### Problem

Altering the speed of the day/night cycle (`time_speed` setting) is one of the most important things to do for many games. Many games want to stop the day/night cycle entirely (`time_speed=0`) but there's a problem.

The problem is the only way to do that is by changing the `time_speed` setting. This "works", but it causes a LOT of sneaky problems (see #14084).
A workaround is for games to manually set the timeofday to a fixed value every X seconds, but this causes visible and annoying sun/moon shaking. And this only works if you want to *halt* the cycle. If you just want to *slow down* the cycle, you're out of luck.

### Solutions

Add a function `minetest.set_time_speed(time_speed)` to set the `time_speed` programmatically. This affects the `time_speed` for the current world only and does not change the configuration files at all.

Calling this function with a number has the same effect as if the `time_speed` setting has been set to that number, except it doesn't affect settings.
Calling this function with `nil` will apply the `time_speed` value from the Minetest settings again.
So the function essentially works as an override.

The game will always be initialied with the `time_speed` value read from `minetest.conf` (or the default). So if the function is never called, Minetest will behave as usual.

This means it is the game's responsibility to manually restore the desired `time_speed` after every launch. I chose this variant for simplicity.

### Alternatives

Introduce temporary changing of settings as suggested in #14084. However, this change is WAY more complex and we'll likely not going to see it anytime soon. Frankly, I believe pretty much `time_speed` is the ONLY reason why #14084 was brought up in the first place.

**Comment by rubenwardy** [comment]
Alternative 3 is #6966

**Comment by Wuzzy2** [comment]
Yes, indeed but maybe a whole rework of the settings system may not be neccessary at all, just a rework of the few *specific* settings that cause problem. I wonder what will be simpler at the end. Feel free to discuss.
